### DISCUSSION TITLE
Highlight marker on click

### DISCUSSION THREAD

**Discussion by kmurari103** [iTwin/itwinjs-core#2934] [Q&A]
Hi there,
I have a list of devices in table and each device in row corresponds a marker in the viewport. so when a row is clicked/ or marker in the viewport is clicked, i want to either change the marker image or highlight that marker to convey that the respective marker is selected. I have large number of devices in table meaning large number of markers to display so clicking on any one marker does not distinguish it from the others and its hard to tell which marker was clicked upon.
Any help on this is appreciated.
Thanks!

**Comment by pmconne** [comment]
You can change how a Marker draws by overriding its [drawDecoration](https://www.itwinjs.org/reference/imodeljs-frontend/rendering/canvasdecoration/drawdecoration/) method.
For an example of how to increase the size of the marker, see [Marker.drawHilited](https://github.com/iTwin/itwinjs-core/blob/580f849174ebe7bdb9f221228d018192527be8f7/core/frontend/src/Marker.ts#L171). This is the effect that is applied while the marker is "hilited" (moused-over). Note it draws 25% larger with a blue shadow.
![image](https://user-images.githubusercontent.com/22944042/146536507-949747c3-e8fc-415a-8831-c8f9146e86ff.png)

For an example of a Marker that draws differently while it is selected, see [SectionMarker](https://github.com/iTwin/itwinjs-core/blob/580f849174ebe7bdb9f221228d018192527be8f7/core/hypermodeling/src/SectionMarkers.ts#L75). After a SectionMarker is clicked, it becomes "active" (selected) and while it is active it draws differently, with a gold shadow.
![image](https://user-images.githubusercontent.com/22944042/146536559-9c9cd6a6-4810-4b44-ad5f-af1036d6f5cc.png)

Both examples achieve their effects by modifying the CanvasRenderingContext2D supplied to drawDecoration.
